src/units/abilities.cpp

```diff
@@ -1573,6 +1617,133 @@ bool attack_type::has_special_or_ability(const std::string& special, bool specia
 }
 //end of emulate weapon special functions.
 
+//function for filter abilities or specials
+namespace {//helper for attack_type::special_matches_filter() and unit::ability_matches_filter()
```

Comment:
Is there a reason that all these were moved over from `unit.cpp`? If nothing else, this will cause a conflict once #7924 is merged.

Reply:
moved reverted